spec/confirm.ts

```diff
@@ -0,0 +1,127 @@
+import assert = require('assert');
+import chai = require('chai');
+import chaiHttp = require('chai-http');
+import { User } from '../src/db/index';
+import { IUser } from '../src/interfaces';
+import { confirm } from '../src/services/userService';
+import { decodeToken, getToken } from '../src/utils';
+import app from './index';
+
+const should = chai.should();
+chai.use(chaiHttp);
+
+describe('Confirm user logic', () => {
+  const user = {
+    email: 'pupkin@mail.com',
+    interfaceLang: 'en',
+    name: 'Ivan',
+    password: 'Real Man'
+  };
+
+  beforeEach(async () => {
+    const newcomer = new User(user);
+    await newcomer.save(err => {
+      return err;
+    });
+  });
+
+  it('should set confirmed field to "true"', async () => {
+    await confirm(user.email);
+    await User.findOne({ email: user.email }, (err, res) => {
+      assert.equal(true, res.confirmed);
+    });
+  });
+
+  it('sould be succes if user exist', done => {
+    chai
+      .request(app)
+      .post('/api/user/confirm')
+      .set('content-type', 'application/json')
+      .send({
+        token:
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InB1cGtpbkBtYWlsLmNvbSJ9.Si2nol4q-7SeMzCkyvm94s45CzP7kx3jG4y9OLdGcv4'
+      })
+      .end((err, res) => {
+        res.should.have.status(200);
+        assert.equal(
+          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb25maXJtZWQiOnRydWUsImVtYWlsIjoicHVwa2luQG1haWwuY29tIiwiaW1hZ2UiOiIiLCJpbnRlcmZhY2VMYW5nIjoiZW4iLCJuYW1lIjoiSXZhbiIsInN1YnNjcmlwdGlvbiI6dHJ1ZX0.gpgESzsa4rQJdHp4RvQ76U-ezVaFTVhyxjpSG8GqiOA',
```

Comment:
Weird hardcoded tokens? Can we generate the token here? Or at least use a constant.
+ is this going to work in 100% of scenarios? If I'm not mistaken the token contains the date when it was generated + the expiration date.